tests/integration/client_v2_test.py

```diff
@@ -69,3 +69,72 @@ def test_whisper_v2(client_v2, data_dir, output_mode, mode, input_file):
                 unified_diff(exp.splitlines(), extracted_text.splitlines(), fromfile="Expected", tofile="Extracted")
             )
             pytest.fail(f"Texts are not similar enough: {similarity * 100:.2f}% similarity. Diff:\n{diff}")
+
+
+@pytest.mark.parametrize(
+    "output_mode, mode, url, input_file, page_count",
+    [
+        ("layout_preserving", "native_text", "https://unstractpocstorage.blob.core.windows.net/public/Amex.pdf",
+         "credit_card.pdf", 7),
+        ("layout_preserving", "low_cost", "https://unstractpocstorage.blob.core.windows.net/public/Amex.pdf",
+         "credit_card.pdf", 7),
+        (
+                "layout_preserving", "high_quality",
+                "https://unstractpocstorage.blob.core.windows.net/public/scanned_bill.pdf",
+                "restaurant_invoice_photo.pdf", 1),
+        ("layout_preserving", "form", "https://unstractpocstorage.blob.core.windows.net/public/scanned_form.pdf",
+         "handwritten-form.pdf", 1),
+    ]
+)
+def test_whisper_v2_url_in_post(client_v2, data_dir, output_mode, mode, url, input_file, page_count):
+    usage_before = client_v2.get_usage_info()
+    whisper_result = client_v2.whisper(
+        mode=mode, output_mode=output_mode, url=url, wait_for_completion=True
+    )
+    logger.debug(f"Result for '{output_mode}', '{mode}', " f"'{input_file}: {whisper_result}")
+
+    exp_basename = f"{Path(input_file).stem}.{mode}.{output_mode}.txt"
+    exp_file = os.path.join(data_dir, "expected", exp_basename)
+    with open(exp_file, encoding="utf-8") as f:
+        exp = f.read()
+
+    assert isinstance(whisper_result, dict)
+    assert whisper_result["status_code"] == 200
+
+    # For text based processing, perform a strict match
+    if mode == "native_text" and output_mode == "text":
+        assert whisper_result["extraction"]["result_text"] == exp
+    # For OCR based processing, perform a fuzzy match
+    else:
+        extracted_text = whisper_result["extraction"]["result_text"]
+        similarity = SequenceMatcher(None, extracted_text, exp).ratio()
+        threshold = 0.97
+
+        if similarity < threshold:
+            diff = "\n".join(
+                unified_diff(exp.splitlines(), extracted_text.splitlines(), fromfile="Expected", tofile="Extracted")
+            )
+            pytest.fail(f"Texts are not similar enough: {similarity * 100:.2f}% similarity. Diff:\n{diff}")
+
+    usage_after = client_v2.get_usage_info()
+    # Verify usage after extraction
+    verify_usage(usage_before, usage_after, page_count, mode)
+
+
+def verify_usage(before_extract, after_extract, page_count, mode='form'):
```

Comment:
NIT: @nagesh-zip this verification might fail if the same API key is used concurrently in different tests - should be okay for now but it can lead to test case failures later